### PR TITLE
Fix issues #339 and #455 (stereographic projections)

### PR DESCRIPTION
**Pull request by perrette**
- added scale_factor to Stereographic projection.
- warnings when true_scale_latitude is used with non-polar
  stereographic projections
- test_stereographic extended

See Issues #455 (and #339) for details.

### CODE REVIEW

File: lib/cartopy/crs.py [hunk 1]
```diff
@@ -1039,12 +1039,24 @@ def y_limits(self):
 class Stereographic(Projection):
     def __init__(self, central_latitude=0.0, central_longitude=0.0,
                  false_easting=0.0, false_northing=0.0,
-                 true_scale_latitude=None, globe=None):
+                 true_scale_latitude=None, 
+                 scale_factor=None, # equivalent to 1.0
```
Comment (QuLogic) [inline lib/cartopy/crs.py]: Info like this comment should be in the docstring (though admittedly, there isn't one.)

File: lib/cartopy/crs.py [hunk 2]
```diff
@@ -1039,12 +1039,24 @@ def y_limits(self):
...
+                 globe=None):
         proj4_params = [('proj', 'stere'), ('lat_0', central_latitude),
                         ('lon_0', central_longitude),
                         ('x_0', false_easting), ('y_0', false_northing)]
+
         if true_scale_latitude:
+            if central_latitude not in (-90., 90.):
+                warnings.warn('"true_scale_latitude" parameter is only used for polar stereographic projections. Consider the use of "scale_factor" instead.')
```
Comment (QuLogic) [inline lib/cartopy/crs.py]: This line is too long.

File: lib/cartopy/crs.py [hunk 3]
```diff
@@ -1039,12 +1039,24 @@ def y_limits(self):
...
             proj4_params.append(('lat_ts', true_scale_latitude))
+
+        # See https://github.com/SciTools/cartopy/issues/455
```
Comment (QuLogic) [inline lib/cartopy/crs.py]: See _what_ from it? It's rather long.

File: lib/cartopy/crs.py [hunk 4]
```diff
@@ -1039,12 +1039,24 @@ def y_limits(self):
...
+        if scale_factor:
+            if true_scale_latitude is not None:
+                warnings.warn('It does not make sense to provide both "scale_factor" and "true_scale_latitude"')
```
Comment (QuLogic) [inline lib/cartopy/crs.py]: Your code still applies both, so what happens when you do?

File: lib/cartopy/tests/crs/test_stereographic.py [hunk 5]
```diff
@@ -52,14 +52,39 @@ def test_eccentric_globe(self):
...
+        globe = ccrs.Globe(ellipse='sphere')
+        stereo = ccrs.NorthPolarStereo(true_scale_latitude=30, globe=globe)
+        expected = ('+ellps=sphere +proj=stere +lat_0=90 +lon_0=0.0 '
+                    '+x_0=0.0 +y_0=0.0 +lat_ts=30 +no_defs')
+        assert_equal(stereo.proj4_init, expected)
+
+    def test_scale_factor(self):
+        # See #455
+        # Use spherical Earth in North Polar Stereographic to check equivalence 
+        # with between true_scale and scale_factor.
```
Comment (QuLogic) [inline lib/cartopy/tests/crs/test_stereographic.py]: s/with//

**Comment by QuLogic** [comment]
@perrette Ping?

**Comment by pelson** [comment]
Looks like we have stalled on this PR. I'll close, but it seems there is no reason we can't accept this PR - just needs a little more work.

**Pull request closed without merging**